src/PresentationalComponents/TagsToolbar/ManageTags.js

```diff
@@ -1,36 +1,113 @@
-    handleModalToggle: () => undefined
+    useEffect(() => {
+        (async () => {
+            setTags(await fetchTags(perPage, page, null, debouncedSearchText));
+            setLoaded(true);
+        })();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [perPage, page]);
+
+    useEffect(() => {
+        setSelected(selectedTags.length ? selectedTags.map(id => ({ id, selected: true })) : []);
+    }, [selectedTags, setSelected]);
+
+    return <TagModal
+        systemName={intl.formatMessage(messages.tagsModalTitle)}
+        {...loaded && {
+            loaded,
+            pagination: {
+                perPage,
+                page,
+                count: totalTags
+            },
+            rows: tags.map(tag => ({
+                id: tag.id,
+                selected: selected.find(selection => selection.id === tag.id),
+                cells: [tag.key, tag.value, tag.namespace]
+            }))
+        }}
+        loaded={ loaded }
+        width='auto'
+        isOpen={ isOpen }
+        toggleModal={() => {
+            setPerPage(10);
+            setPage(1);
+            setSearchText();
+            setSelected([]);
+            toggleModal();
+        }}
+        filters={[
+            {
+                label: intl.formatMessage(messages.filterTagsInModal),
+                placeholder: intl.formatMessage(messages.filterTagsInModal),
+                value: 'tags-filter',
+                filterValues: {
+                    key: 'text-filter',
+                    onChange: (event, value) => setSearchText(value),
+                    value: searchText
+                }
+            }
+        ]}
+        onUpdateData={(data) => {
+            setLoaded(false);
+            setPerPage(data.perPage);
+            setPage(data.page);
+        }}
+        columns={ [
+            { title: intl.formatMessage(messages.tagsModalName) },
+            { title: intl.formatMessage(messages.tagsModalValue) },
+            { title: intl.formatMessage(messages.tagsModalSources) }
+        ] }
+        selected={selected}
+        onSelect={(selected) => setSelected(selected.map(tag => ({ id: tag.id, selected: true })))}
+        onApply={() => {
+            setSelectedTags(selected.map(tag => tag.id));
+            setSelected([]);
+        }}
+        tableProps={{ canSelectAll: false }}
+    />;
 };
 
-const mapStateToProps = ({ AdvisorStore, ownProps }) => ({
-    selectedTags: AdvisorStore.selectedTags,
-    ...ownProps
-});
+ManageTags.propTypes = {
+    toggleModal: PropTypes.func.isRequired,
+    fetchTags: PropTypes.func.isRequired,
+    totalTags: PropTypes.number.isRequired,
+    isOpen: PropTypes.bool.isRequired,
+    selectedTags: PropTypes.array,
+    setSelectedTags: PropTypes.func,
+    intl: PropTypes.any
+};
 
-export default injectIntl(connect(mapStateToProps)(ViewHostAcks));
+export default injectIntl(connect(
+    state => ({ selectedTags: state.AdvisorStore.selectedTags }),
```

Comment:
OOO OOOO deconstruct Advisorstore!

```suggestion
    ({AdvisorStore}) => ({ selectedTags: AdvisorStore.selectedTags }),
```